src/Compilers/CSharp/Portable/Syntax/SyntaxEquivalence.cs

```diff
@@ -117,7 +121,8 @@ private static bool AreEquivalentRecursive(GreenNode? before, GreenNode? after,
             }
             finally
             {
-                stack.Free();
+                stack.Clear();
+                s_equivalenceCheckStack.Free(stack);
```

Comment:
is there not FreeAndClear/ClearAndFree?

Reply:
That's defined at the workspace level

Reply:
well that's silly :)

Reply:
I would be fine to move that down if you wanted.

Reply:
Though I will point out that those helpers stop returning items after they get to 512 items. That might not work well here either.